### PR TITLE
Not sending background attributionsrc request if no registrar is supported

### PR DESCRIPTION
**Pull request by linnan-github**
***
<a href="https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/782.html" title="Last updated on May 4, 2023, 6:03 PM UTC (c2cf742)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/782/ad3fcc2...linnan-github:c2cf742.html" title="Last updated on May 4, 2023, 6:03 PM UTC (c2cf742)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1110,6 +1110,16 @@ An <dfn>eligibility</dfn> is one of the following:
 
 </dl>
 
+A <dfn>registrar</dfn> is one of the following:
+
+<dl dfn-for="registrar">
```
Comment (apasel422) [inline index.bs:1115]: Can we remove the `support key` definition below if we're going to add this definition? They are identical.

File: index.bs [hunk 2]
```diff
@@ -2917,6 +2929,14 @@ To <dfn noexport>get an OS-registration URL from a header list</dfn> given a
 1. If |value| is not a [=string=], return null.
 1. Return the result of running the [=URL parser=] on |value|.
 
+To <dfn>get supported registrars</dfn>:
+1. Let |supportedRegistrars| be an [=set/is empty|empty=] [=ordered set=].
```
Comment (apasel422) [inline index.bs:2933]:
```suggestion

1. Let |supportedRegistrars| be an [=set/is empty|empty=] [=ordered set=].
```

**Review by apasel422 — DISMISSED** [review]

File: index.bs [hunk 3]
```diff
@@ -2917,29 +2929,31 @@ To <dfn noexport>get an OS-registration URL from a header list</dfn> given a
...
+1. Let |supportedRegistrars| be the result of [=getting supported registrars=].
 1. Let |dict| be an [=ordered map=].
-1. If the user agent supports web registrations, [=map/set=]
-    |dict|["<code>[=support key/web=]</code>"] to true.
-1. If the user agent supports OS registrations, [=map/set=]
-    |dict|["<code>[=support key/os=]</code>"] to true.
+1. If |supportedRegistrars| [=set/contains=] "<code>[=registrar/web=]</code>", [=map/set=]
+    |dict|["<code>[=registrar/web=]</code>"] to true.
+1. If |supportedRegistrars| [=set/contains=] "<code>[=registrar/os=]</code>", [=map/set=]
+    |dict|["<code>[=registrar/os=]</code>"] to true.
```
Comment (apasel422) [inline index.bs:2956]: To avoid listing each possible registrar explicitly, maybe just iterate over `supportedRegistrars`.

Comment (linnan-github) [reply index.bs]: Done, thanks.

**Review by apasel422 — APPROVED** [review]

**Pull request merged**